### ISSUE TITLE
[Issue]: Follow/Notes updates to My Grants

### ISSUE THREAD

**Issue by ClaireValdivia** [usdigitalresponse/usdr-gost#3601]
### Why is this issue important?

We want the My Grants tab to reflect the new activity introduced with Follow & Notes

### Current State

On the My Grants page, there are currently four "tabs" across the top to filter the grants that display in the table:
- Shared With Your Team - shows any grants shared with your team
- Interested - shows grants that the user has marked as one of the Interested statuses
- Not Applying - shows grants that the user has marked as one of the Not Applying statuses
- Applied - shows grants that the user has marked as one of the Applied statuses

### Expected State

- "Shared With Your Team" should be updated to "Shared With My Team" and will display the same grants it currently does (i.e. any grants shared with your team)
- A new tab will be created that is called "Followed by My Team" - this should be a filtered list of all grants currently followed by the user and/or any other user in their team.
- The current tabs for Interested, Not Applying, Applied should be removed
- The grants table column "Interested Teams" should be renamed "Followed By"
- Both "Followed by" and "Viewed by" should display the full team name if any user in the user's team has followed the grant (rather than current state of showing abbreviation). Team name text should wrap if needed.
- If more than one team member follows or views a grant, the team name should only appear once in the respective column
- If other teams view or follow the grant, all team names where someone follows the grant should display in comma separated list (i.e. US Digital Response, State of Nevada, Department of Education)
- Export to CSV - the exported file should reflect the renamed column of "Followed By", and the full team name(s) should display in a comma separated list for both Viewed by and Followed by.

### Implementation Plan

Use the followNotesEnabled feature flag - https://github.com/usdigitalresponse/usdr-gost/issues/3405

**Comment by phm200** [comment]
@ClaireValdivia I assume the URL should reflect the new tab and new tab name when that tab is selected?

i.e.
- "Shared With My Team" URL would be `/my-grants/shared-with-my-team`
- "Followed by My Team" URL would be `/my-grants/followed-by-my-team`

**Comment by ClaireValdivia** [comment]
@phm200 yes, thanks for catching that!

**Comment by phm200** [comment]
rolling along here; need to make sure the wrap behavior is right: 
`Team name text should wrap if needed`

and make sure the export csv is displaying the followed by data:
`Update CSV export to support updated column label/values based on feature flag (route /grants/exportCSVNew)`

hopefully get this up for review by end of weekend

**Comment by phm200** [comment]
@ClaireValdivia should the changes to the Grants table, like renaming the "Interested Teams" column to "Followed By" also reflect in the main "Browse Grants" page? As well as the updated Export to CSV output? Or should these changes only be done to "My Grants" page?

I'm finishing up this PR and noticed in testing that the changes I made for "My Grants" page also reflect in "Browse Grants" and want to confirm if thats intended or not

**Comment by ClaireValdivia** [comment]
@phm200 yes, confirming that the changes should reflect in the table on both the My Grants and Browse Grants pages.

And yes, the changes should be reflected in teh exports to csv

**Comment by ClaireValdivia** [comment]
The My Grants tabs and table columns look great. 

Going to open a new issue for parts of this issue that were not finished:
- Both "Followed by" and "Viewed by" should display the full team name if any user in the user's team has followed the grant (rather than current state of showing abbreviation). Team name text should wrap if needed.
- If more than one team member follows or views a grant, the team name should only appear once in the respective column
- If other teams view or follow the grant, all team names where someone follows the grant should display in comma separated list (i.e. US Digital Response, State of Nevada, Department of Education)
- Export to CSV - the exported file should reflect the renamed column of "Followed By", and the full team name(s) should display in a comma separated list for both Viewed by and Followed by.

**Comment by ClaireValdivia** [comment]
outstanding work is captured in this issue: https://github.com/usdigitalresponse/usdr-gost/issues/3855
